cartographer_ros/src/cartographer_node_main.cc

```diff
@@ -225,6 +251,28 @@ Rigid3d Node::LookupToTrackingTransformOrThrow(
                                  ros::Duration(kMaxTransformDelaySeconds)));
 }
 
+void Node::OdomMessageCallback(const nav_msgs::Odometry::ConstPtr& msg) {
+  auto sensor_data = ::cartographer::common::make_unique<SensorData>(
+    msg->header.frame_id, ToRigid3d(msg->pose.pose), ToPoseCovariance(msg->pose.covariance));
+
+  sensor_collator_.AddSensorData(
+    kTrajectoryId,
+    ::cartographer::common::ToUniversal(FromRos(msg->header.stamp)),
+    odom_subscriber_.getTopic(), std::move(sensor_data));
+
+}
+
+void Node::AddOdom(
+    int64 timestamp, const string& frame_id,
+    const Rigid3d odom, const PoseCovariance covar) {
```

Comment:
Did you intend to pass by const reference? Then this should be "const Rigid3d& odometry" and "const PoseCovariance& covariance".